### PR TITLE
[sockpp] Add new port

### LINKED ISSUE #10971: [New Port Request] sockpp
Library name: sockpp

Library description: Simple, modern, C++ socket library.

Source repository URL: https://github.com/fpagliughi/sockpp

Project homepage (if different from the source repository):

Anything else that is useful to know when adding (such as optional features the library may have that should be included):

### PR DESCRIPTION
**Pull request by thewisebro**
**Describe the pull request**

- What does your PR fix? Fixes #10971

- Which triplets are supported/not supported? Have you updated the CI baseline? None of the uwp triplets are supported, rest all are supported. Yes.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)? Yes

### CODE REVIEW

File: ports/sockpp/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,34 @@
+include(vcpkg_common_functions)
```
Comment (LilyWangL) [inline ports/sockpp/portfile.cmake:1]: `include(vcpkg_common_functions)` is deprecated. Please remove it.

File: ports/sockpp/CONTROL [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Source: sockpp
+Version: 0.7
+Homepage: https://github.com/fpagliughi/sockpp
+Description: Simple, modern, C++ socket library. This is a fairly low-level C++ wrapper around the Berkeley sockets library using socket, acceptor, and connector classes that are familiar concepts from other languages.
```
Comment (LilyWangL) [inline ports/sockpp/CONTROL:4]: Can you please add [Supports](https://github.com/microsoft/vcpkg/blob/e62d1361288e83eba786395b60361ab35ba83800/docs/maintainers/control-files.md#supports) for this port in the CONTROL file?

File: ports/sockpp/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        -DSOCKPP_BUILD_SHARED=OFF
+        -DSOCKPP_BUILD_STATIC=ON
+        -DSOCKPP_BUILD_DOCUMENTATION=OFF
+        -DSOCKPP_BUILD_EXAMPLES=OFF
+        -DSOCKPP_BUILD_TESTS=OFF
+)
+
+vcpkg_install_cmake()
+
+file(COPY ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright)
```
Comment (LilyWangL) [inline ports/sockpp/portfile.cmake:29]: Can you please use command `file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)` to handle copyright?

**Comment by LilyWangL** [comment]
Can you please add the following command in _scripts/ci.baseline.txt_?
```
sockpp:arm-uwp=fail
sockpp:x64-uwp=fail
```

**Review by LilyWangL — CHANGES_REQUESTED** [review]

File: ports/sockpp/CONTROL (hunk 2 above)
Comment (thewisebro) [reply ports/sockpp/CONTROL:4]: Ack.

File: ports/sockpp/portfile.cmake (hunk 1 above)
Comment (thewisebro) [reply ports/sockpp/portfile.cmake:1]: Ack.

File: ports/sockpp/portfile.cmake (hunk 3 above)
Comment (thewisebro) [reply ports/sockpp/portfile.cmake:29]: Ack.

**Comment by thewisebro** [comment]
> Can you please add the following command in _scripts/ci.baseline.txt_?
> 
> ```
> sockpp:arm-uwp=fail
> sockpp:x64-uwp=fail
> ```

Updated _scripts/ci.baseline.txt_

File: ports/sockpp/sockppConfig.cmake [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    if(SOCKPP_FIND_REQUIRED)
+        set(SOCKPP_FIND_REQUIRED_sockpp TRUE)
+    endif()
+    set(SOCKPP_FOUND TRUE)
+endif()
+
+set(SOCKPP_INCLUDE_DIRS ${_DIR}/include)
+set(SOCKPP_LIBRARIES)
+if (EXISTS ${_DIR}/lib/libsockpp.so)
+    list(APPEND SOCKPP_LIBRARIES ${_DIR}/lib/libsockpp.so)
```
Comment (dan-shaw) [inline ports/sockpp/sockppConfig.cmake:13]: It looks like this config doesn't support Windows, can you add that in `Supports` in the Control file.

File: ports/sockpp/usage [hunk 5]
```diff
@@ -0,0 +1,5 @@
+The package sockpp provides CMake integration:
+
+    find_package(sockpp CONFIG REQUIRED)
+    target_include_directories(main PRIVATE ${SOCKPP_INCLUDE_DIRS})
+    target_link_libraries(main PRIVATE ${SOCKPP_LIBRARIES})
```
Comment (dan-shaw) [inline ports/sockpp/usage:5]: Usage instructions should be automatically generated.

**Review by dan-shaw — APPROVED** [review]

File: ports/sockpp/sockppConfig.cmake (hunk 4 above)
Comment (thewisebro) [reply ports/sockpp/sockppConfig.cmake:13]: This was an oversight. I've added support for Windows as well.

File: ports/sockpp/usage (hunk 5 above)
Comment (thewisebro) [reply ports/sockpp/usage:5]: Sorry, but how to enable that? Its not being generated by default.

**Pull request merged**